File: 0002.js
solve = n => {
  let prevNum = 1, currNum = 2;
  let total = 0;

  while (currNum < n) {
    if (currNum % 2 === 0) {
      total += currNum;
    }

    [prevNum, currNum] = [currNum, prevNum + currNum]
  }

  return total;
}

console.log(solve(4000000));

File: 0002.rb
def solve n
  total, prev_num, curr_num = 0, 1, 2

  while curr_num < n do
    if curr_num % 2 === 0
      total += curr_num
    end

    prev_num, curr_num = curr_num, prev_num + curr_num
  end

  total
end

puts solve 4000000